eth/filters/api.go

```diff
@@ -348,7 +350,15 @@ func (api *FilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([]*type
 			end = crit.ToBlock.Int64()
 		}
 		// Construct the range filter
-		filter = api.sys.NewRangeFilter(begin, end, crit.Addresses, crit.Topics)
+		var err error
+		filter, err = api.sys.NewRangeFilter(begin, end, crit.Addresses, crit.Topics)
+		if err != nil {
+			// Compatible with old implement
+			if err == ErrInvalidBlockRange {
+				return nil, nil
+			}
```

Comment:
IMO this is overly cautious. Let's just return an error on nonsense-requests (and then we can remove `ErrInvalidBlockRange` as a public error, just inline it where we use it